app/src/main/java/app/sram/bikestore/activity/BikeStoreListUseCase.kt

```diff
@@ -16,20 +16,20 @@ import javax.inject.Inject
 class BikeStoreListUseCase @Inject constructor(
     private val bikeRepo: BikeRepo,
     @IOScheduler private val ioScheduler: Scheduler,
-    @UIScheduler private val mainScheduler: Scheduler
+    @UIScheduler private val mainScheduler: Scheduler,
+    private val entityToItemMapper: EntityToItemMapper
 ) {
-    fun execute(param: Param): Single<ResultModel<List<BikeStoreItem>>> {
-        return bikeRepo.list(param).map { mapToAdapterModel(it, param.location) }.subscribeOn(ioScheduler)
-            .observeOn(mainScheduler)
+    fun execute(location: ScramLocation, refresh: Boolean): Single<ResultModel<List<BikeStoreItem>>> {
+        return Single.fromCallable { bikeRepo.list(location, refresh) }.subscribeOn(ioScheduler)
+            .map { mapToAdapterModel(it, location) }.observeOn(mainScheduler)
     }
 
     private fun mapToAdapterModel(resultModel: ResultModel<BikeStoreData>, deviceLocation: ScramLocation):
         ResultModel<List<BikeStoreItem>> {
-
             return when (resultModel) {
                 is Success -> Success(
                     resultModel.data.list.map {
-                        entityToItem(it, deviceLocation.calculate(it.location))
+                        entityToItemMapper.mapToItem(it, deviceLocation.calculate(it.location))
```

Comment:
Change static mapper function to mapper class. 
It also can use the D to bind an interface with an object.